tests/e2e/testsLibrary/che-code/ProjectAndFileTestsCheCode.ts

```diff
@@ -10,24 +10,103 @@
 
 @injectable()
 export class ProjectAndFileTestsCheCode {
 
+    private static readonly ENDPOINTS_TITLE_TEXT: string = 'endpoints';
+    private static readonly TRUST_AUTHORS_DIALOG_BOX_TEXT: string = 'Do you trust the authors of the files in this folder?';
+    private static readonly TRUST_AUTHORS_DIALOG_BUTTON_TEXT: string = 'Yes, I trust the authors';
+
     constructor(
         @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) {}
 
-    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<void> {
+    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<Workbench> {
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
         try {
             await this.driverHelper.getDriver().wait(until.elementLocated(By.className('monaco-workbench')), TimeoutConstants.TS_SELENIUM_START_WORKSPACE_TIMEOUT);
+            const workbench: Workbench = new Workbench();
+            const sideBarView: SideBarView = workbench.getSideBar();
+            const sidebarViewContent: ViewContent = sideBarView.getContent();
+            let sidebarViewSections: ViewSection[];
+            while (true) {
+                sidebarViewSections = await sidebarViewContent.getSections();
+                Logger.debug(`ProjectAndFileTestsCheCode.waitWorkspaceReadinessForCheCodeEditor - Waiting for ${ProjectAndFileTestsCheCode.ENDPOINTS_TITLE_TEXT} sidebar section:`);
+                for (const section of sidebarViewSections) {
+                    Logger.trace(`Sidebar section: ${await section.getTitle()}`);
+                    if (await section.getTitle() === ProjectAndFileTestsCheCode.ENDPOINTS_TITLE_TEXT) {
+                        return workbench;
+                    }
+                }
+                await this.driverHelper.getDriver().sleep(polling);
+            }
         } catch (err) {
             Logger.error(`ProjectAndFileTestsCheCode.waitWorkspaceReadinessForCheCodeEditor - waiting for workspace readiness failed: ${err}`);
             throw err;
         }
     }
+
+    public async waitAndConfirmTrustAuthorsDialogBox(): Promise<void> {
+        Logger.debug('ProjectAndFileTestsCheCode.waitAndConfirmTrustAuthorsDialogBox');
+        try {
+            Logger.trace('ProjectAndFileTestsCheCode.waitAndConfirmTrustAuthorsDialogBox - waiting for trust authors dialog box overlay');
+            const dialog: WebElement = await this.driverHelper.waitPresence(By.id('monaco-dialog-message-text'));
+            const dialogText: string = await dialog.getText();
+            if (dialogText === ProjectAndFileTestsCheCode.TRUST_AUTHORS_DIALOG_BOX_TEXT) {
+                const trustAuthorsDialogCheckbox: WebElement = await this.driverHelper.waitPresence(By.css('body > div > div.monaco-dialog-modal-block.dimmed > div > div > div.dialog-message-row > div.dialog-message-container > div.dialog-checkbox-row > div.monaco-custom-toggle.codicon.codicon-check.monaco-checkbox'));
```

Comment:
The code looks like VS Code editor page object, with specific locators and logic, which are duplicating https://github.com/redhat-developer/vscode-extension-tester/tree/main/locators/lib and 

It doesn't comply with decision to use vscode-extension-tester external library and so don't spend time for support of VS Code editor specific logic and locators.

Reply:
I am sure we can make this locator smaller using xpath